packages/periphery/contracts/oracles/PendleOracle.sol

```diff
@@ -0,0 +1,143 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity 0.8.19;
+
+import '@openzeppelin/contracts/access/Ownable2Step.sol';
+import '@openzeppelin/contracts/utils/math/Math.sol';
+import '@pendle/core-v2/contracts/oracles/PendlePtLpOracle.sol';
+import '@pendle/core-v2/contracts/core/Market/v3/PendleMarketV3.sol';
+import '@marginly/contracts/contracts/interfaces/IPriceOracle.sol';
+
+/// @notice Oracle for Pendle PT / underlying token, e.g. PT-ezETH-27JUN2024 / WETH
+contract PendleOracle is IPriceOracle, Ownable2Step {
+  struct OracleParams {
```

Comment:
Seems like there is no need to store `ptDecimals` and `syDecimals`,the `PRICE_DECIMALS + poolParams.ptDecimals - poolParams.syDecimals` is sufficient enough (or even `10 ** (PRICE_DECIMALS + poolParams.ptDecimals - poolParams.syDecimals)` since 160 * 3 + 16 * 2 = 512)